src/btree/bt_walk.c

```diff
@@ -40,21 +40,15 @@ retry:	WT_INTL_INDEX_GET(session, ref->home, pindex);
 	 * is slower.
 	 */
 	i = ref->pindex_hint;
-	if (i < pindex->entries && pindex->index[i] == ref) {
-		*pindexp = pindex;
-		*slotp = i;
-		return;
-	}
+	if (i < pindex->entries && pindex->index[i] == ref)
+		goto found;
 	while (++i < pindex->entries)
-		if (pindex->index[i] == ref) {
-			*pindexp = pindex;
-			*slotp = ref->pindex_hint = i;
-			return;
-		}
+		if (pindex->index[i] == ref)
+			goto found;
 	for (i = 0; i < pindex->entries; ++i)
 		if (pindex->index[i] == ref) {
-			*pindexp = pindex;
-			*slotp = ref->pindex_hint = i;
+found:			*pindexp = pindex;
```

Comment:
Yes, as @keithbostic, we're already doing that at https://github.com/wiredtiger/wiredtiger/blob/develop/src/btree/bt_walk.c#L437-L439

tbh, since we already set the hint during splits, I don't know if we should ever update it during reads.